### ISSUE TITLE
Regarding reproduction of full MVD result.

### ISSUE THREAD

**Issue by yc-song** [Noen61/MVD#1]
Hi. Thanks for great work!

Regarding reproducing the result from your checkpoint, I have a couple of questions.

0. Do you support saving candidates for reranking over candidates retrieved by MVD candidates in this repo?

1. If I run **mvd.bin** which is accessible from the link (which correspons to[ this)](https://drive.google.com/file/d/17DOtfKwSCjS9kZsDFG0lQQ0kghXXtE1u/view?usp=sharing) at the section named 'MVD Training', is it expected to get the result as follows?:

Model | Recall@1 | Recall@2 | Recall@4 | Recall@8 | Recall@16 | Recall@32 | Recall@50 | Recall@64
-- | -- | -- | -- | -- | -- | -- | -- | --
MVD | 52.51 | 64.77 | 73.43 | 79.74 | 84.35 | 88.17 | 90.43 | 91.55

I ran following command from scripts/run_evaluate.sh where I changed --pretrain_retriever into mvd.bin and got result closer to MVD w/o global view, which shows slightly lower performance than MVD. 

command:
python -m torch.distributed.launch --nproc_per_node=1 main.py \
--do_eval \
--kb zeshel \
--task_name retriever \
--data_dir data/zeshel \
--entity_data_dir data/zeshel/entity \
--output_dir output/retriever \
--bert_model bert-base-uncased \
**--pretrain_retriever output/retriever/mvd.bin** \
--max_seq_length 128 \
--infer_view_type local \
--train_view_type local \
--max_ent_length 512 \
--max_view_length 40 \
--max_view_num 10 \
--eval_batch_size 1024 \
--num_train_epochs 40 \
--top_k 100 \
--faiss \
--seed 10000 \
--do_lower_case \

result:
```
  "eval_metrics": {
    "global_step": 0,
    "top1_hits": 0.5123,
    "top2_hits": 0.6327,
    "top4_hits": 0.7176,
    "top8_hits": 0.7802,
    "top10_hits": 0.7973,
    "top16_hits": 0.8296,
    "top30_hits": 0.8643,
    "top32_hits": 0.8677,
    "top50_hits": 0.8902,
    "top64_hits": 0.9026,
    "top100_hits": 0.92
  },
```

**Comment by Noen61** [comment]
0. run **scripts/run_predict.sh** script to generate candidates for reranking
1. set **infer_view_type = global-local** in command


> Hi. Thanks for great work!
> 
> Regarding reproducing the result from your checkpoint, I have a couple of questions.
> 
> 0. Do you support saving candidates for reranking over candidates retrieved by MVD candidates in this repo?
> 1. If I run **mvd.bin** which is accessible from the link (which correspons to[ this)](https://drive.google.com/file/d/17DOtfKwSCjS9kZsDFG0lQQ0kghXXtE1u/view?usp=sharing) at the section named 'MVD Training', is it expected to get the result as follows?:
> 
> Model	Recall@1	Recall@2	Recall@4	Recall@8	Recall@16	Recall@32	Recall@50	Recall@64
> MVD	52.51	64.77	73.43	79.74	84.35	88.17	90.43	91.55
> I ran following command from scripts/run_evaluate.sh where I changed --pretrain_retriever into mvd.bin and got result closer to MVD w/o global view, which shows slightly lower performance than MVD.
> 
> command: python -m torch.distributed.launch --nproc_per_node=1 main.py --do_eval --kb zeshel --task_name retriever --data_dir data/zeshel --entity_data_dir data/zeshel/entity --output_dir output/retriever --bert_model bert-base-uncased **--pretrain_retriever output/retriever/mvd.bin** --max_seq_length 128 --infer_view_type local --train_view_type local --max_ent_length 512 --max_view_length 40 --max_view_num 10 --eval_batch_size 1024 --num_train_epochs 40 --top_k 100 --faiss --seed 10000 --do_lower_case \
> 
> result:
> 
> ```
>   "eval_metrics": {
>     "global_step": 0,
>     "top1_hits": 0.5123,
>     "top2_hits": 0.6327,
>     "top4_hits": 0.7176,
>     "top8_hits": 0.7802,
>     "top10_hits": 0.7973,
>     "top16_hits": 0.8296,
>     "top30_hits": 0.8643,
>     "top32_hits": 0.8677,
>     "top50_hits": 0.8902,
>     "top64_hits": 0.9026,
>     "top100_hits": 0.92
>   },
> ```

**Comment by yc-song** [comment]
Thanks for reply!
